### ISSUE TITLE
Usage of openapi-generator 6.0.0-SNAPSHOT

### ISSUE THREAD

**Issue by AndreasSchultzGEA** [adewg/ICAR#295]
**Setup**: 
- artefacts are not public available, therefore download sources and compile
- compiles only with Java JDK 8. Set Path to JDK and also JAVA_HOME

**generated Java-Sources**:
- introduction of IcarAnimalEventCoreResource causes Problems
- single discriminator located in IcarResource - also defined as dedicated property
`@JsonTypeName("icarResource")
@Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2022-04-25T07:58:00.892882900+02:00[Europe/Berlin]")
public class IcarResource {

  **@JsonProperty("resourceType")
  private String resourceType;**
`

**JSON**:
- ReST-call - e.g. animals, parturitions - contains double definition of 'resourceType'
{
    "view": null,
    "member": [
        {
            "resourceType": "icarReproParturitionEventResource",
            "resourceType": null,
            "@self": null,

**Comment by cookeac** [comment]
Hi @AndreasSchultzGEA as you've built openapi-generator 6.0.0 beta from the sources, perhaps you could check **modules/openapi-generator/src/main/resources/JavaSpring/typeInfoAnnotation.mustache**  - there should be some additional code in there that was not in previous versions which deals with `@JsonIgnoreProperties`. That will indicate to us whether the code to fix the bug has made it into the release.

The code is supposed to:
- suppress the additional declaration that would come from declaring the property in both `"properties"` and as a `"discriminator"`, 
- Set the property during deserialization (as is happening with the first of your duplicate properties) and 
- ignore changes made other than in serialization/deserialization (so it behaves as a discriminator and not a general use property).

**Comment by AndreasSchultzGEA** [comment]
There are changes about the discriminator in the master, but not in the branch 6.0.x.
Is there anyone we could trigger about this, @cookeac ?
But as I compiled the master - before the branch 6.0.x has been available - the problem seem still to be present.
I will re-check this before our next meeting.
@AlexeyHardCode : Do you have any results so far?

**Comment by cookeac** [comment]
Hi @AndreasSchultzGEA do you know what the value (true/false) for `legacyDiscriminatorBehavior` is within the open API configuration? 

There is something about this here: https://openapi-generator.tech/docs/generators/openapi/

**Comment by AndreasSchultzGEA** [comment]
I discovered, that we use a patched version of the **typeInfoAnnotation.mustache** mentioned above. Therefore, I executed the complete test again without the patched template.

The result in concerns of the new "resourceType" is now ok. Defined once, output contains it once and with reasonable content.

But we also generate the ReST-interfaces from the definitions at the "url-schmes"-folder. And there happens sth. very strange. Each (!) interface-definitions contain a duplicate definition of the HTTP-200-Result-handler as there is a "default"-response defined for each endpoint within the files at url-schemes. 
And the Java-class configured for handling this event seem to be randomly defined, sometimes even does it not exist which leads to compile-errors.

![image](https://user-images.githubusercontent.com/79915286/169006369-96417d00-790c-4c99-aaa5-8fe34615e492.png)

If the default-handler is removed from the files in url-schemes, the code is ok again and works as expected.

**Comment by cookeac** [comment]
We agreed that now #299 has been logged, this issue itself is resolved.

**Comment by AndreasSchultzGEA** [comment]
The re-check of the migration to openapi-geneartor 6.0.0-SNAPSHOT causes again some problems as described above: Out of the default-error-handling a random default-error class is generated. The naming seems to suite to sth. from the first endpoint within the urlscheme-file under investigation.

We found a way to suppress this behaviour by not defining the default-(error)-response inline in each url-scheme, but to define it explicitly as following in each scheme-file:

```
    "responses": {
      "default": {
        "description": "An error has occured while handling the request. Check the content of the message for the error details.",
        "content": {
          "application/json": {
            "schema": {
              "$ref": "../collections/icarErrorCollection.json"
            }
          }
        }
      }
    },
```
This implies to add an new object icarErrorCollection which may have to following content:
```
{
  "type": "object",
  "properties": {
    "errors": {
      "type": "array",
      "items": {
        "$ref": "../resources/icarResponseMessageResource.json"
      }
    }
  },
  "description": "A default response containing only a list of errors"
}
```

Benefits: The definition of the default-response is cleaned up and references a resource instead an inline definition.
Merge-request will follow - if we agree on this change.

**Comment by AndreasSchultzGEA** [comment]
This fits together with the changes made by Graham in https://github.com/adewg/ICAR/pull/302
  ```
      "responses": {
          "200": {
            "description": "Successful. The response contains the diagnoses for the given location.",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/icarMilkingVisitEventCollection"
                }
              }
            }
          },
          "400": {
            "$ref": "#/components/responses/default"
          },
          "500": {
            "$ref": "#/components/responses/default"
          }
        }

```

**Comment by cookeac** [comment]
@AndreasSchultzGEA I agree, this seems like a good solution which is cleaner and also more aligned with how we returned errors for batch posts.

**Comment by cookeac** [comment]
Opened in the context of #299

**Comment by cookeac** [comment]
We agreed that this should be done.

**Comment by cookeac** [comment]
Resolved again by #321
